zipline/data/us_equity_pricing.py

```diff
@@ -119,54 +104,112 @@ class NoDataOnDate(Exception):
     pass
 
 
-class BcolzDailyBarWriter(with_metaclass(ABCMeta)):
+def check_uint32_safe(value, colname):
+    if value >= UINT32_MAX:
+        raise ValueError(
+            "Value %s from column '%s' is too large" % (value, colname)
+        )
+
+
+def winsorise_uint32(df, column, *columns):
+    """Drops any record where a value would not fit into a uint32.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The dataframe to winsorise.
+    *columns : iterable[str]
+        The names of the columns to check.
+
+    Returns
+    -------
+    truncated : pd.DataFrame
+        ``df`` with rows that contain values that do not fit into a uint32
+        zeroed out.
+    """
+    columns = (column,) + columns
+    df[reduce(op.or_, (df[c] > UINT32_MAX for c in columns))] = 0
```

Comment:
^ that should in theory be more performant as well, though it's pandas, so who knows.